projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -677,6 +686,18 @@ function getParamName(
   return undefined;
 }
 
+function getClassName(constructorParameter: ts.Node): string | undefined {
```

Comment:
Nice!

If you want to have it more concise, we can do it like this:

```ts
function getClassName(constructorParameter: ts.Node): string | undefined {
  const identifierNode = constructorParameter
    .getChildren()
    .find((node) => node.kind === ts.SyntaxKind.TypeReference)
    ?.getChildren()
    .find((node) => node.kind === ts.SyntaxKind.Identifier);

  return identifierNode ? identifierNode.getText() : undefined;
}
```

But it's also fine as it is.